### DISCUSSION TITLE
Livewire form not working with vue added

### DISCUSSION THREAD

**Discussion by chistel** [livewire/livewire#4088] [1. Help]
I have a project which was built Vue, but am gradually trying to get rid of Vue.

my app.js has the necessary live wire/vue setup, but my forms don't work. when I inspect elements, the form does not carry the submit event. but when I remove Vue from app.js, the form works.

a snippet of the app.js

`
    import Vue from 'vue'
    import 'livewire-vue'
    
    Vue.prototype.$http = window.axios;
    Vue.prototype.EventBus = new Vue();
    window.Vue = Vue;
    
    new Vue({
		    el: '#app',
		    data() {
			    return {
				    component: '',
				    modalIds: {},
				    showPageLoader: false,
				    // options
				    breakpoints: {
					    sm: 540,
					    md: 720,
					    lg: 960,
					    xl: 1140
				    },
				    body: $('body'),
				    sidebar: $('.sidebar')
			    }
		    },
    });
`

a snippet of the form

`
<x-form wire:submit.prevent="saveMaterial" method="POST">
			
			<div class="row me-1 pe-0">
				<div class="col-md-5 mb-3">
					<x-form-label label="School" class="form-label mandatory"/>
					<x-form-select name="school_id">
						<option value="be">Laspotech</option>
						<option value="2">CU</option>
					</x-form-select>
				</div>
				<div class="col-md-7 mb-3">
					<x-form-label label="School course" class="form-label mandatory"/>
					<x-form-select name="course_id">
						<option value="1">Maths</option>
						<option value="2">English language</option>
					</x-form-select>
				</div>
			</div>
			<div class="row">
				<div class="col-md-8 mb-3">
					<x-form-label label="Course title" class="form-label mandatory" file="title"/>
					<x-form-input name="true" :show-errors="true" id="material_course_title" name="title"
									  placeholder="Course title"/>
				</div>
				<div class="col-md-2 mb-3">
					<x-form-label label="Price" class="form-label mandatory" for="course_price"/>
					<x-form-input class="shadow-none" id="course_price" name="price"
									  placeholder="Course price"/>
				</div>
				<div class="col-md-2 mb-3">
					<x-form-label label="Enable" class="form-label" for="course_enabled">
						@slot('help')
							<a href="javascript:void(0)" class="text-info fa fa-question-circle" data-bs-toggle="popover"
								title="Enable to allow users purchase your material"
								data-bs-content="Enable to allow users purchase your material"></a>
						@endslot

					</x-form-label>
					<x-form-checkbox name="enable" switch="true" id="course_enabled" value="1"/>
				</div>
			</div>

			<div class="col-md-12 mb-3">
				<x-form-label label="Enable" class="form-label required" for="material_course_file">
					File
				</x-form-label>
				<div class="border border-1 rounded px-5 py-4 position-relative d-flex align-items-center justify-content-center overflow-hidden" x-data="{dropingFile: false}" x-on:drop="dropingFile = false"
					  x-on:dragover.prevent="dropingFile = true"
					  x-on:dragleave.prevent="dropingFile = false">
					<span class="flex-shrink-0 border border-1 border-primary rounded-2 text-uppercase px-6 py-2 bg-primary text-white me-4">
						Choose files
					</span>
					<span class="lh-1 overflow-hidden">
						or drag and drop files here
					</span>
					<input class="position-absolute left-0 top-0 h-100 w-100 o-0 copy-docs" type="file">
				</div>
			</div>
			
			<x-form-submit>
				<span class="text-green-500">Send</span>
			</x-form-submit>
		</x-form>
`

**Comment by joshhanley** [comment]
@chistel I personally haven't used Vue and Livewire together, so can't be of much help.

@godismyjudge95, sorry to tag you (saw you were using Vue due to your PR🙂). Any chance you might be able to assist with this?

**Comment by godismyjudge95** [comment]
@chistel I believe you need to put this line:
```
window.Vue = Vue;
```

Above this line:
```
import 'livewire-vue'
```

In the end it would look like so:
```
import Vue from 'vue'
window.Vue = Vue
import 'livewire-vue'

Vue.prototype.$http = window.axios;
Vue.prototype.EventBus = new Vue();
```

Let me know if that doesn't work.

**Reply by chistel** [reply]
> @chistel I believe you need to put this line:
> ```
> window.Vue = Vue;
> ```
> 
> Above this line:
> ```
> import 'livewire-vue'
> ```
> 
> In the end it would look like so:
> ```
> import Vue from 'vue'
> window.Vue = Vue
> import 'livewire-vue'
> 
> Vue.prototype.$http = window.axios;
> Vue.prototype.EventBus = new Vue();
> ```
> 
> Let me know if that doesn't work.


Thanks, would try it out tomorrow.

**Reply by joshhanley** [reply]
@godismyjudge95 legend! Thanks for helping out 🙂
